Doc: document that JXL-in-GeoTIFF requires tiled=yes (#11110)
OSGeo/gdal: doc/source/drivers/raster/gtiff.rst

      * ``LERC_ZSTD`` is available when ``LERC`` and ``ZSTD`` are available.

      * ``JXL`` is for JPEG-XL, and is only available when using internal libtiff and building GDAL against

>         https://github.com/libjxl/libjxl . JXL compression should be used with the ``TILED=YES`` creation

Reviewer: ```suggestion
        https://github.com/libjxl/libjxl . It is recommended to use JXL compression with the ``TILED=YES`` creation
```
